internal/settings/providers/netcup/client.go

```diff
@@ -0,0 +1,171 @@
+	}
+	request, err := http.NewRequestWithContext(ctx, http.MethodPost, c.endpoint, bytes.NewBuffer(b))
+	if err != nil {
+		return nil, fmt.Errorf("%w: %s", errors.ErrBadRequest, err)
+	}
+	headers.SetUserAgent(request)
+	response, err := c.client.Do(request)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %s", errors.ErrUnsuccessfulResponse, err)
+	}
+	defer response.Body.Close()
+
+	b, err = io.ReadAll(response.Body)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %s", errors.ErrUnmarshalResponse, err)
+	}
+
+	var res Response
+	err = json.Unmarshal(b, &res)
+	if err != nil {
+		return nil, err
+	}
+
+	if res.isError() {
+		return nil, errors.ErrBadHTTPStatus
+	}
+
+	return &res, nil
+}
+
+func (c *Client) Login(ctx context.Context) error {
+	var params = NewParams()
+	params.AddParam("apikey", c.APIKey)
+	params.AddParam("apipassword", c.Password)
+	params.AddParam("customernumber", c.CustomerNumber)
+
+	request := NewNetcupRequest("login", &params)
+
+	response, err := c.do(ctx, request)
+	if err != nil {
+		return err
+	}
+
+	var loginResponse LoginResponse
+	err = json.Unmarshal(response.ResponseData, &loginResponse)
+
+	switch {
+	case err != nil:
+		return err
+	case loginResponse.Session == "":
+		return errors.ErrNoSession
+	default:
+		c.Session = loginResponse.Session
+	}
+
+	return nil
+}
+
+func (c *Client) InfoDNSRecords(ctx context.Context, domainname string) (*DNSRecordSet, error) {
+	params, err := c.addAuthParams(domainname)
+	if err != nil {
+		return nil, err
+	}
+
+	request := NewNetcupRequest("infoDnsRecords", params)
+
+	response, err := c.do(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+
+	var dnsRecordSet DNSRecordSet
+	err = json.Unmarshal(response.ResponseData, &dnsRecordSet)
+	if err != nil {
+		return nil, err
+	}
+
+	return &dnsRecordSet, nil
+}
+
+func (c *Client) UpdateDNSRecords(ctx context.Context, domain string, records *DNSRecordSet) (*Response, error) {
+	params, err := c.addAuthParams(domain)
+	if err != nil {
+		return nil, err
+	}
+
+	params.AddParam("dnsrecordset", records)
+	request := NewNetcupRequest("updateDnsRecords", params)
+
+	response, err := c.do(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+
+	return response, nil
+}
+
+func (c *Client) addAuthParams(domainname string) (*Params, error) {
+	if c.Session == "" {
+		return nil, errors.ErrNoSession
+	}
+
+	params := NewParams()
+	params.AddParam("apikey", c.APIKey)
+	params.AddParam("apisessionid", c.Session)
+	params.AddParam("customernumber", c.CustomerNumber)
+	params.AddParam("domainname", domainname)
+
+	return &params, nil
+}
+
+func (c *Client) GetRecordToUpdate(ctx context.Context, domain, host string, ip net.IP) (*DNSRecord, error) {
+	records, err := c.InfoDNSRecords(ctx, domain)
+	if err != nil {
+		return nil, err
+	}
+
+	recordType := constants.A
+	if ip.To4() == nil {
+		recordType = constants.AAAA
+	}
+	if records.GetRecordOccurences(host, recordType) > 1 {
+		return nil, errors.ErrListRecords
+	}
```

Comment:
List the number of occurrences you got and what you expect (1) in the error context using `fmt.Errorf("%w: context blabla", errors.ErrListRecords)`